ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/filebased/FileManager.java

```diff
@@ -69,4 +76,25 @@ public void appendRecord(String file, String recordToAppend) {
 			createNewRecords(file, Stream.of(recordToAppend));
 		}
 	}
+
+	public void destroyState() {
+		try {
+			Path statePath = Path.of(stateDirectory).toAbsolutePath();
+			try (Stream<Path> list = Files.list(statePath)) {
+				list.forEach(path -> {
+					try {
+						Files.delete(path);
+					} catch (IOException e) {
+						throw new DestroyStateFailedException(e);
+					}
+				});
+				Files.delete(statePath);
+				Files.delete(statePath.getParent());
+			} catch (NoSuchFileException ignored) {
```

Comment:
Can we avoid this by using "Files.deleteIfExists" ?

Reply:
Will check. This however is existing code :)